File: app/src/main/java/com/tt/handsomeman/adapter/JobAdapter.java
package com.tt.handsomeman.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.tt.handsomeman.R;
import com.tt.handsomeman.databinding.ItemJobBinding;
import com.tt.handsomeman.model.Job;

import java.text.ParseException;
import java.util.List;

public class JobAdapter extends RecyclerView.Adapter<JobAdapter.JobViewHolder> {

    private List<Job> jobList;
    private LayoutInflater layoutInflater;
    private Context context;
    private ItemJobBinding binding;

    private OnItemClickListener mListener;

    public JobAdapter(Context context, List<Job> jobList) {
        this.jobList = jobList;
        this.context = context;
        layoutInflater = LayoutInflater.from(context);
    }

    public void setOnItemClickListener(OnItemClickListener listener) {
        mListener = listener;
    }

    @NonNull
    @Override
    public JobViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        binding = ItemJobBinding.inflate(layoutInflater, parent, false);
        View item = binding.getRoot();
        return new JobViewHolder(item, mListener);
    }

    @Override
    public void onBindViewHolder(@NonNull JobViewHolder holder, int position) {
        Job job = jobList.get(position);

        holder.tvJobTitle.setText(job.getTitle());
        try {
            holder.tvCreateTime.setText(context.getResources().getQuantityString(R.plurals.numberOfHour, job.setCreateTimeBinding(), job.setCreateTimeBinding()));
            holder.tvDeadline.setText(context.getResources().getQuantityString(R.plurals.numberOfDay, job.setDeadlineBinding(), job.setDeadlineBinding()));
        } catch (ParseException e) {
            e.printStackTrace();
        }
        holder.tvBudget.setText(job.setBudgetRange());
        holder.tvLocationName.setText(job.getLocation());
    }

    @Override
    public int getItemCount() {
        return jobList.size();
    }

    public interface OnItemClickListener {
        void onItemClick(int position);
    }

    class JobViewHolder extends RecyclerView.ViewHolder {

        private final TextView tvJobTitle, tvCreateTime, tvBudget, tvLocationName, tvDeadline;
        private final ImageButton btnJobDetail;

        JobViewHolder(@NonNull View itemView, final OnItemClickListener listener) {
            super(itemView);

            tvJobTitle = binding.textViewJobName;
            tvCreateTime = binding.textViewCreateTime;
            tvBudget = binding.textViewBudgetRange;
            tvLocationName = binding.textViewLocationName;
            tvDeadline = binding.textViewDeadLine;

            btnJobDetail = itemView.findViewById(R.id.imageButtonJobDetail);

            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    if (listener != null) {
                        int position = getAdapterPosition();
                        if (position != RecyclerView.NO_POSITION) {
                            listener.onItemClick(position);
                        }
                    }
                }
            });

            btnJobDetail.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    if (listener != null) {
                        int position = getAdapterPosition();
                        if (position != RecyclerView.NO_POSITION) {
                            listener.onItemClick(position);
                        }
                    }
                }
            });
        }
    }
}

File: app/src/main/java/com/tt/handsomeman/adapter/JobAdapter.kt
package com.tt.handsomeman.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageButton
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

import com.tt.handsomeman.R
import com.tt.handsomeman.model.Job

class JobAdapter(private val context: Context, private val jobList: List<Job>) : RecyclerView.Adapter<JobAdapter.MyViewHolder>() {

    private var mListener: OnItemClickListener? = null
    private val layoutInflater: LayoutInflater

    init {
        layoutInflater = LayoutInflater.from(context)
    }

    interface OnItemClickListener {
        fun onItemClick(position: Int)
    }

    fun setOnItemClickListener(listener: OnItemClickListener) {
        mListener = listener
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val item = layoutInflater.inflate(R.layout.item_job, parent, false)
        return MyViewHolder(item, mListener)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val job = jobList[position]

        holder.tvJobTitle.text = job.title
        holder.tvCreateTime.text = job.setCreateTimeBinding(job.createTime)
        holder.tvBudget.text = job.setBudgetRange(job.budgetMin, job.budgetMax)
        holder.tvLocationName.text = job.setLocationBinding(job.location!!)
        holder.tvDeadline.text = job.setDeadlineBinding(job.deadline)
    }

    override fun getItemCount(): Int {
        return jobList.size
    }

    internal inner class MyViewHolder(itemView: View, listener: OnItemClickListener?) : RecyclerView.ViewHolder(itemView) {

        var tvJobTitle: TextView
        var tvCreateTime: TextView
        var tvBudget: TextView
        var tvLocationName: TextView
        var tvDeadline: TextView
        var btnJobDetail: ImageButton

        init {

            tvJobTitle = itemView.findViewById(R.id.textViewJobName)
            tvCreateTime = itemView.findViewById(R.id.textViewCreateTime)
            tvBudget = itemView.findViewById(R.id.textViewBudgetRange)
            tvLocationName = itemView.findViewById(R.id.textViewLocationName)
            tvDeadline = itemView.findViewById(R.id.textViewDeadLine)

            btnJobDetail = itemView.findViewById(R.id.imageButtonJobDetail)

            itemView.setOnClickListener {
                if (listener != null) {
                    val position = adapterPosition
                    if (position != RecyclerView.NO_POSITION) {
                        listener.onItemClick(position)
                    }
                }
            }

            btnJobDetail.setOnClickListener {
                if (listener != null) {
                    val position = adapterPosition
                    if (position != RecyclerView.NO_POSITION) {
                        listener.onItemClick(position)
                    }
                }
            }
        }
    }
}